### PR TITLE
Basic Exporter Completed

### PR DESCRIPTION
**Pull request by wasd243**
This PR completes the initial exporter pipeline, including Bridge integration, ProseMirror JSON interpreter support (`[[include]]` and themes are NOT completed yet), caching layer, and architecture refactoring.

The system now supports a working end-to-end export flow and basic wikitext WYSIWYG and generate.

This is a stable baseline for future improvements and refactoring.

Future work:
- Edge case handling
- Performance optimization
- Further cleanup of bridge and interpreter layers
- Complete TODOs
- Complete themes
- Tests

It's finally works, sounds good but still have a lot of things in TODO.

### CODE REVIEW

File: crates/ltmf/src/ftml_fmt/fmt_newline.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+/// This function collapses multiple newlines into two.
+fn collapse_newlines(ftml: &str) -> String {
+    let mut output = String::with_capacity(ftml.len());
+    let mut newline_count = 0;
+
+    for character in ftml.chars() {
+        if character == '\n' {
+            if newline_count < 2 {
+                output.push(character);
+            }
```
[the remark this answers is not in the document]
Comment (wasd243) [reply crates/ltmf/src/ftml_fmt/fmt_newline.rs:14]: NOT PLANNED

File: crates/ltmf/src/ftml_fmt/add_user_css.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+pub(super) fn add_user_css(ftml: &str, user_css: Option<&str>) -> String {
+    match user_css {
+        Some(user_css) => format!("[[module css]]{user_css}[[/module]]\n\n{ftml}"),
```
[the remark this answers is not in the document]
Comment (wasd243) [reply crates/ltmf/src/ftml_fmt/add_user_css.rs:3]: TODO
